test/physics/em/Rayleigh.test.cc

```diff
@@ -33,46 +46,151 @@ class RayleighInteractorTest : public celeritas_test::InteractorHostTestBase
     {
         ASSERT_TRUE(interaction);
 
-        // Check change to parent track
-        EXPECT_GT(this->particle_track().energy().value(),
+        // Check change to parent track - coherent scattering
+        EXPECT_EQ(this->particle_track().energy().value(),
                   interaction.energy.value());
-        EXPECT_LT(0, interaction.energy.value());
-        EXPECT_SOFT_EQ(1.0, celeritas::norm(interaction.direction));
         EXPECT_EQ(celeritas::Action::scattered, interaction.action);
-
-        // XXX Check secondaries
-
-        // Check conservation between primary and secondaries
-        this->check_conservation(interaction);
     }
 
   protected:
-    celeritas::detail::RayleighInteractorPointers pointers_;
+    std::shared_ptr<RayleighModel>       model_;
+    celeritas::detail::RayleighNativeRef group_;
 };
 
 //---------------------------------------------------------------------------//
 // TESTS
 //---------------------------------------------------------------------------//
 
-TEST_F(RayleighInteractorTest, basic) {}
+TEST_F(RayleighInteractorTest, basic)
+{
+    // Sample an element (TODO: add ElementSelector)
+    ElementId el_id{0};
+
+    std::vector<double>            angle;
+    std::vector<unsigned long int> rng_counts;
+
+    // Sample scattering angle and count rng used for each incident energy
+    for (double inc_e : {0.05, 0.1, 1.0, 10.0, 100.0})
+    {
+        // Set the incident particle energy
+        this->set_inc_particle(pdg::gamma(), MevEnergy{inc_e});
+
+        // Create the interactor
+        RayleighInteractor interact(this->model_->host_group(),
+                                    this->particle_track(),
+                                    this->direction(),
+                                    el_id);
+
+        RandomEngine& rng_engine = this->rng();
+
+        // Produce a sample from the original/incident photon
+        celeritas::Interaction result = interact(rng_engine);
+        SCOPED_TRACE(result);
+        this->sanity_check(result);
+        rng_engine.count();
+
+        angle.push_back(dot_product(result.direction, this->direction()));
+        rng_counts.push_back(rng_engine.count());
+    }
+
+    const double expected_angle[] = {-0.726848858395344,
+                                     -0.95887836792401,
+                                     -0.910276761243175,
+                                     -0.223090692143936,
+                                     -0.388188923742863};
+
+    const unsigned long int expected_rng_counts[]
+        = {17774ul, 19342ul, 19416ul, 19424ul, 19438ul};
+
+    EXPECT_VEC_SOFT_EQ(expected_angle, angle);
+    EXPECT_VEC_EQ(expected_rng_counts, rng_counts);
+}
+
+TEST_F(RayleighInteractorTest, stress_test)
+{
+    const int num_samples = 1028;
+
+    // Sample an element
+    ElementId el_id{0};
+
+    std::vector<real_type>         average_angle;
+    std::vector<unsigned long int> average_rng_counts;
```

Comment:
Since this is an average I'd make it a `double`.

Reply:
Changed to real_type.